
From: NeilBrown <neilb@cse.unsw.edu.au>

Thanks to Al Viro for noticing that putrootfh could return either a linux or
an nfs error.

Signed-off-by: J. Bruce Fields <bfields@citi.umich.edu>
Signed-off-by: Neil Brown <neilb@cse.unsw.edu.au>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/fs/nfsd/nfs4proc.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN fs/nfsd/nfs4proc.c~nfsd4-fix-putrootfh-return fs/nfsd/nfs4proc.c
--- 25/fs/nfsd/nfs4proc.c~nfsd4-fix-putrootfh-return	2004-09-23 22:06:49.021636200 -0700
+++ 25-akpm/fs/nfsd/nfs4proc.c	2004-09-23 22:06:49.025635592 -0700
@@ -267,7 +267,7 @@ nfsd4_putrootfh(struct svc_rqst *rqstp, 
 	status = exp_pseudoroot(rqstp->rq_client, current_fh,
 			      &rqstp->rq_chandle);
 	if (!status)
-		status = nfsd_setuser(rqstp, current_fh->fh_export);
+		status = nfserrno(nfsd_setuser(rqstp, current_fh->fh_export));
 	return status;
 }
 
_
